com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/SetBreakpointsRequestHandler.java

```diff
@@ -77,6 +87,12 @@
             return AdapterUtils.createAsyncErrorResponse(response, ErrorCode.SET_BREAKPOINT_FAILURE,
                         String.format("Failed to setBreakpoint. Reason: '%s' is an invalid path.", bpArguments.source.path));
         }
+
+        if (bpArguments.sourceModified) {
+            IHotCodeReplaceProvider hcrProvider = context.getProvider(IHotCodeReplaceProvider.class);
+            hcrProvider.completed().thenAccept((List<String> result) -> reinstallBreakpoints(result));
```

Comment:
for sourceModified case, the BreakpointManager actually will reinstallBreakpoint, is there any possible to cause race condition?

Reply:
yep, what's why i add clone in reinstallBreakpoints previously